### PR TITLE
Add supply query demo

### PR DESCRIPTION
**Pull request by AlexKubiesa**

### CODE REVIEW

**Review by nightroman — APPROVED** [review]
I left some comments on Slack, feel free to follow.

File: SupplyQueryDemo/SupplyQueryDemo/Program.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        Query = Query,
+        Variables = new Dictionary<string, object> { { "mpn", mpn } }
+    };
+
+    Response result = await supplyClient.RunQueryAsync(request);
+
+    if (result.Data?.SupSearchMpn?.Results == null)
+    {
+        continue;
```
Comment (RobAltium) [inline SupplyQueryDemo/SupplyQueryDemo/Program.cs:44]: I'd be tempted to output a line here: `($"Sorry no part found for MPN search: {mpn}")`

Comment (RobAltium) [reply SupplyQueryDemo/SupplyQueryDemo/Program.cs:44]: The other option is to handle errors as returned from the GraphQL response, e.g. something like this...
```
    {
        [JsonProperty("data")]
        public Data Data { get; set; }

        [JsonProperty("errors")]
        public List<Error> Errors { get; set; }
    }

    public partial class Error
    {
        [JsonProperty("message")]
        public string Message { get; set; }
    }
```    
... but that probably adds complexity for what we want here.

File: SupplyQueryDemo/SupplyQueryDemo/SupplyClient.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            return supplyClient;
+        }
+
+        internal static async Task<Response> RunQueryAsync(this HttpClient supplyClient, Request request)
+        {
+            string requestString = JsonConvert.SerializeObject(request);
+            HttpResponseMessage httResponse = await supplyClient.PostAsync(supplyClient.BaseAddress, new StringContent(requestString, Encoding.UTF8, "application/json"));
+            httResponse.EnsureSuccessStatusCode();
+            string responseString = await httResponse.Content.ReadAsStringAsync();
+            Response response = JsonConvert.DeserializeObject<Response>(responseString);
```
Comment (RobAltium) [inline SupplyQueryDemo/SupplyQueryDemo/SupplyClient.cs:54]: Here would be a great place to add a comment with a reference to [https://github.com/NexarDeveloper/nexar-templates/tree/main/nexar-console-supply](https://github.com/NexarDeveloper/nexar-templates/tree/main/nexar-console-supply) for a different, more automatic way of processing the response.

**Review by RobAltium — CHANGES_REQUESTED** [review]
A couple of minor comments.  I'd be happy if we just tidy up comments - add as many more as you think could be useful for new coders/programmers coming into Nexar API.  Happy to defer the error handling until later (or another sample).

**Review by RobAltium — APPROVED** [review]
Looks good.

**Pull request merged**